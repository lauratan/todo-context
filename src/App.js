import React, { useState, useReducer } from 'react';
import {Container} from 'react-bootstrap';
import AddTodo from './components/AddTodo';
import TodoList from './components/TodoList';

const App = () => {
  const initialState = {
    todos: [
      {
        id: 1,
        task: 'Walk dog',
        completed: false
      },
      {
        id: 2,
        task: 'Cook dinner',
        completed: true
      }
    ]
  };

  const [todos, setTodo] = useState(initialState.todos);

  const addTodo = (todo) => {
    setTodo([...todos, todo]);
  }

  return (
    <Container className="m-3">
      <AddTodo addTodo={addTodo}/>
      <TodoList todos={todos} taskCount={todos.length}/>
    </Container>
  )
}

export default App;