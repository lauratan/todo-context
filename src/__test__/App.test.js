import "@testing-library/jest-dom";
import React             from 'react';
import {render, cleanup} from "@testing-library/react";
import App               from '../App';

afterEach(cleanup);

describe('App component', ()=> {
  it("matches snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    render(<App/>)  
  });
})