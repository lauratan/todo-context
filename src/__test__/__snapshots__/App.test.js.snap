// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-3 container"
  >
    <div
      class="mb-2"
    >
      <form
        class=""
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
          >
            New Todo
          </label>
          <input
            class="form-control"
            placeholder="e.g Water plants"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Add New Todo
        </button>
      </form>
    </div>
    <div>
      <div
        class="todo-heading"
      >
        <h2>
          Todo List: 
        </h2>
        <p
          class="counter m-1"
        >
          2 Tasks
        </p>
      </div>
      <hr />
      <ul
        class="todo-list"
      >
        <li
          class="todo-item incomplete"
        >
          <input
            class="mr-2"
            type="checkbox"
          />
           Walk dog 
          <i
            class="far fa-trash-alt"
          />
        </li>
        <li
          class="todo-item completed"
        >
          <input
            checked=""
            class="mr-2"
            type="checkbox"
          />
           Cook dinner 
          <i
            class="far fa-trash-alt"
          />
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
